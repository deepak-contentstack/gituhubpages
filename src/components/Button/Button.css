@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/common.css';

[type='button']:not(:disabled),
[type='reset']:not(:disabled),
[type='submit']:not(:disabled),
button:not(:disabled) {
  	cursor: pointer;
}

.ui-component button{
  	margin: 0 calc(var(--space-2) + 1px);
}

/* Button */

.Button {
	background-color: transparent;
	border: 1px solid transparent;
	border-radius: $radii-6;
	font-family: $font-family-primary;
	font-size: $size-font-medium;
	font-weight: $font-weight-bold;
	line-height: $space-12;
	/* line-height: calc(var(--space-12) + 1px); */
	min-height: $space-32;
	min-width: $space-32;
	padding: $space-8 $space-16;
	/* padding: $space-8 $space-16 calc(var(--space-8) + 1px); */
	position: relative;
	text-align: center;
	transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
	border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
	vertical-align: middle;

	&__loader{
		left: 50%;
		position: absolute;
		top: 50%;
		-webkit-transform: translateX(-50%) translateY(-50%);
		-moz-transform: translateX(-50%) translateY(-50%);
	}

	&__icon {
		align-items: center;
		color: $color-base-gray-20;
		display: flex;
		height: 100%;
		justify-content: center;
		left: 0;
		position: absolute;
		top: 0;
		transition: border-color 0.15s ease-in-out;
		width: 2rem;
	}

	&.Button--icon {
		padding: 0;
		svg {
			margin : auto!important;
		}
	}

	&.Button--icon-alignment-right{
		svg:first-child{
		float: right;
		margin-left: 0.625rem;
		margin-right: 0;
		}
	}

	&:focus {
		box-shadow: 0px 0px 0px 4px #BBB4F4;
		outline: none;
	}

	&--state {
		&-disabled {
			background-color: $color-brand-secondary-lighter !important;
			border: 1px solid transparent;
			color: $color-font-tertiary !important;
			opacity: $opacity-5;
			pointer-events: none;
			text-shadow: none !important;
			& svg path {
				fill: $color-font-tertiary !important
			}
		}
		&-loading{
			cursor: default;
			pointer-events: none;
		}
	}

	&--style {
		&-circle {
			border-radius: $radii-50;
			padding: 0;
		}

		&-fullwidth {
			width: 100%;
		}
	}

	&--primary {
		background-color: $color-brand-primary-base;
		color: $color-font-white;
		text-shadow: 0 1px 0 rgba(0, 0, 0, 0.16);

		&:hover {
			background-color: $color-brand-primary-darker;
		}

		&:focus {
			box-shadow: 0px 0px 0px 4px #BBB4F4;
		}

		&:active {
			background-color: $color-brand-primary-dark;
		}

		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		}
	}

	&--secondary {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-link;
		color: $color-font-link;
		
		&:hover{
			border:1px solid $color-base-purple-25;
			color:$color-base-purple-25;
		}

		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		}

		&:hover {
			border : 1px solid $color-base-purple-25;
			color: $color-brand-primary-darker;
		}

		&:focus {
			box-shadow: 0px 0px 0px 4px #BBB4F4;
		}

		&:active {
			border: 1px solid $color-brand-primary-dark;
			color: $color-brand-primary-dark;
		}
	}

	&--tertiary {
		color: $color-font-link;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		
		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
			path {
				fill: $color-font-link;
			}
		}
	}

	&--tertiary-outline {
		background: $color-brand-white-base;
		border: .0625rem solid $color-base-gray-5;
		color: $color-font-base;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		height: 2.5rem;
		padding: .4375rem .625rem;

		&:hover {
			border: .0625rem solid $color-border-link;
			color: $color-font-link;

			svg:first-child {
				path, circle, rect {
					fill: $color-font-link;
					stroke: $color-font-link;
				}
			}
		}
		
		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		} 
	}

	&--outline {
		background: $color-brand-white-base;
		border: 1px solid $color-border-dark;
		color: $color-font-base;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;

		&:hover {
			border: 1px solid $color-border-link;
			color: $color-font-link;
		}
	}

	&--success {
		background-color: $color-brand-white-base;
		color: $color-font-success;
		&:hover {
			background-color: $color-brand-success-base;
			border: 1px solid $color-base-white-5;
			color: $color-font-white;
		}
	}

	&--control {
		border: 1px solid $color-base-gray-5;
		border-radius: $radii-8;
		color: $color-base-gray-20;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		line-height: 1;
		max-width: 11.563rem;
		overflow: hidden;
		padding: $space-12 $space-12 $space-12 $space-30;
		position: relative;
		text-overflow: ellipsis;
		white-space: nowrap;

		&:hover {
			border-color: $color-brand-primary-base;
			color: $color-brand-primary-base;
			
			svg {
				path {
				stroke: $color-brand-primary-base;
				}
			}
		}
	}

	&--delete {
		border: 1px solid $color-border-warning;
		color: $color-font-warning;
		svg {
			margin: 0 $space-10 0 0;
			vertical-align: top;
			path {
				fill : $color-base-orange-5;
			}
		}
		&:hover {
			background-color: $color-brand-warning-base;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
				}
			}
		}
	}

	&--outline-delete {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-link;
		color: $color-font-link;
		svg {
			margin: 0 .625rem 0 0;
			path {
				fill : $color-base-purple-5;
			}
		}
		&:hover{
			background-color:$color-brand-warning-base;
			border: 1px solid $color-border-warning;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
				}
			}
		}
	}

	&--outline-success {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-success;
		color: $color-font-success;
		svg {
			margin: 0 .625rem 0 0;
			vertical-align: bottom;
		}
		&:hover {
			background-color: $color-brand-success-base;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
					stroke: $color-base-white-5;
				}
			}
		}
	}
	
	&--bulk-delete{
		background-color: $color-brand-primary-base;
		color: $color-font-white;
		svg {
			margin: 0 .625rem 0 0;
			path {
				fill : $color-font-white;
			}
		}
		&:hover{
			background-color: $color-font-white;
			color: $color-base-orange-5;
			svg {
				path {
					fill : $color-base-orange-5;
				}
			}
		}
	}

	&--danger {
		background-color: $color-brand-white-base;
		color: $color-font-warning;
		&:hover {
			background-color: $color-brand-warning-base;
			border: 0.0625rem solid $color-base-white-5;
			color: $color-font-white;
		}
	}

	&--warning {
		background-color: $color-brand-white-base;
		color: $color-font-attention;
		&:hover {
			background-color: $color-brand-attention-base;
			border: 0.0625rem solid $color-base-white-5;
			color: $color-font-white;
		}

		&-link {
			color: $color-base-purple-5;
			font-weight: $font-weight-regular;
			text-decoration: none;
		}
	}

	&--light {
		color: $color-brand-secondary-base;
		&:hover {
			background-color: $color-brand-secondary-lighter;
			color: $color-font-base;
		}
	}

	&--white {
		background: $color-brand-white-base;
		box-shadow: 0 8px 8px rgba(0, 0, 0, 0.02), 0 4px 8px rgba(0, 0, 0, 0.06), 0 8px 16px rgba(0, 0, 0, 0.04);
		color: $color-font-secondary;
		font-weight: $font-weight-regular;
		svg {
			margin-right: $space-10;
		}
		&:hover {
			background: $color-brand-secondary-lighter;
			color: $color-font-black;
			svg path {
				fill : $color-base-gray-20;
			}
		}
	}

	&--size {
		&-large {
			font-size: $size-font-xl;
			padding: $space-12 $space-18 $space-12;
		}

		&-small {
			font-size: $size-font-medium;
			min-height: $space-25;
			padding: $space-5 $space-10;
		}

		&-fullwidth {
			width: 100%;
		}
	}

	&:active {
		border: 1px solid $color-brand-primary-dark;
	}

	.badge {
		margin-left: $space-10;

		&:first-child {
			margin: 0;
			margin-right: $space-10;
		}
	}
}

[type='button']:not(:disabled),
[type='reset']:not(:disabled),
[type='submit']:not(:disabled),
button:not(:disabled) {
  	cursor: pointer;
}

.Button-wrapper button {
  	margin: 0 calc(var(--space-2) + 1px);
}

.Button-container {
	height: 6.25rem;
	margin: $space-15;
	padding: $space-34 0;
	text-align: center;
	width: 12.5rem;
}
