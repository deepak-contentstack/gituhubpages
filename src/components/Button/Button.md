# Button

## Best practices

- Use clear and accurate labels.
- Start button labels with verbs, for example "Add field", not "New field".
- Keep the position of the buttons consistent on the interface.
- Reduce the complexity by using a small number of actions on the interface. It may create confusion if too many actions are used.
