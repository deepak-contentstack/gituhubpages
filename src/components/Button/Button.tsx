import React, {
  Component,
  CSSProperties,
  MouseEvent as ReactMouseEvent,
  MouseEventHandler,
  Fragment,
} from 'react'
import cn from 'classnames'

import withDeprecatedProp from '../../utils/hooks/depricatedPropsHoc';

// import Icon, { IconProps } from '../Icon2'
// import { iconName } from '../Icon2/constants'
// import AsyncLoader from '../AsyncLoader/AsyncLoader'

import './Button.css'

const defaultProps = {
  isFullWidth: false,
  isRounded: false,
  disabled: false,
  buttonType: 'primary',
  type: 'button',
  isLoading: false,
  testId: 'cs-button'
}

type DefaultProps = Readonly<typeof defaultProps>

export type ButtonProps = {
  /**
   * Class names to be appended to the className prop of the Button Component
   */
  className?: string
  children?: React.ReactNode
  /**
   * Pass the icon that want to add in button
   */
  icon?: any
  /**
   * If you want to show only icon, set the param to ‘true’
   */
  onlyIcon?: boolean
  /**
   * Determines the alignment of the icon
   */
  iconAlignment?: 'left' | 'right'
  /**
   * Pass additional props for icon
   */
  // iconProps?: Partial<IconProps>
  /**
   * Pass an id to the button
   */
  id?: string
  /**
   * Determine the type of button you want to use
   */
  buttonType?:
  | 'primary'
  | 'secondary'
  | 'tertiary'
  | 'tertiary-outline'
  | 'outline'
  | 'success'
  | 'danger'
  | 'warning'
  | 'light'
  | 'control'
  | 'delete'
  | 'bulk-delete'
  | 'outline-delete'
  | 'outline-success'
  | 'white'
  /**
   * Determine the size of the button: small or large
   */
  size?: 'small' | 'large'
  /**
   * If button covers full width
   */
  isFullWidth?: boolean
  /**
   * If button is rounded
   */
  isRounded?: boolean
  /**
   * An ID used for testing purposes applied as a data attribute (data-test-id)
   */
  testId?: string
  /**
   * Used to set the button in the loading state
   */
  isLoading?: boolean
  /**
   * Determines the color of the loading animation you want to display
   */
  loadingColor?: string
  /**
   * Action that you want to perform when a button is clicked
   */
  onClick?: MouseEventHandler
  /**
   * Determines whether button is disabled or not
   */
  disabled?: boolean
  /**
   * If you want a button as a link, provide the details in href.
   */
  href?: string
  /**
   * Type of button action (button, submit,reset)
   */
  type?: 'button' | 'submit' | 'reset'
  /**
   * Pass any CSS properties for the button
   */
  style?: CSSProperties
} & Partial<DefaultProps>
export class Button extends Component<ButtonProps> {
  public static propTypes: any = {}
  static defaultProps = defaultProps
  render() {
    const {
      className,
      children,
      buttonType,
      size,
      isFullWidth,
      isRounded,
      onClick,
      disabled,
      href,
      type,
      icon,
      iconAlignment,
      style,
      isLoading,
      loadingColor,
      onlyIcon,
      testId,
      id,
      iconProps = {},
      ...otherProps
    }: any = this.props
    const classNames = cn('Button', className, [`Button--${buttonType}`], {
      [`Button--state-disabled`]: disabled,
      [`Button--size-${size}`]: size,
      ['Button--style-fullwidth']: isFullWidth,
      ['Button--style-circle']: isRounded,
      [`Button--icon-alignment-${iconAlignment}`]: iconAlignment,
      [`Button--state-loading`]: isLoading,
      [`Button--icon`]: onlyIcon,
    })
    const addButtonIconClassNames = cn('Button__icon')
    const Element = href ? 'a' : 'button'
    return (
      <Element
        data-test-id={testId}
        className={classNames}
        disabled={disabled}
        href={!disabled ? href : null}
        id={id}
        aria-label={id || 'aria-button'}
        type={type}
        style={style}
        onClick={(e: any) => {
          if (onClick) {
            onClick(e)
          }
        }}
        {...otherProps}
      >
        <div className="flex-center">
          {isLoading &&
            <div className="Button__loader">
              {/* <AsyncLoader color={loadingColor} /> */}
            </div>
          }
          <div
            style={{ visibility: !isLoading ? 'visible' : 'hidden' }}
            className={iconAlignment === "right" ? "flex-right" : "flex-v-center"}
          >
            {buttonType === 'control' && (
              <span className={addButtonIconClassNames}>
                {/* <Icon icon={iconName.PlusSign} /> */}
              </span>
            )}
            {/* {icon && <Icon icon={icon} {...iconProps} />} */}
            {children && <>{children}</>}
          </div>
        </div>
      </Element>
    )
  }
}

export default withDeprecatedProp(Button, { 'loading': 'isLoading' })