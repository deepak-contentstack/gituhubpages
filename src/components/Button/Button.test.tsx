import React from 'react'
import renderer from 'react-test-renderer'
import EnzymeToJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import Button from './Button'

describe('Button Types', () => {
  it('render Primary Button', () => {
    const Primary = () => {
      return <Button buttonType="primary"></Button>
    }
    const tree = renderer.create(<Primary />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render Tertiary Button', () => {
    const element = mount(<Button buttonType="tertiary" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('tertiary')).toBe(true)
  })

  it('render Secondary Button', () => {
    const element = mount(<Button buttonType="secondary" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('secondary')).toBe(true)
  })

  it('render Warning Button', () => {
    const element = mount(<Button buttonType="warning" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('warning')).toBe(true)
  })

  it('render Danger Button', () => {
    const element = mount(<Button buttonType="danger" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('danger')).toBe(true)
  })

  it('render Success Button', () => {
    const element = mount(<Button buttonType="success" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('success')).toBe(true)
  })

  it('render Light Button', () => {
    const element = mount(<Button buttonType="light" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('light')).toBe(true)
  })

  it('render Control Button', () => {
    const element = mount(<Button buttonType="control" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('control')).toBe(true)
  })

  it('render Delete Button', () => {
    const element = mount(<Button buttonType="delete" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('delete')).toBe(true)
  })

  it('render White Button', () => {
    const element = mount(<Button buttonType="white" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('white')).toBe(true)
  })

  it('render Bulk delete Button', () => {
    const element = mount(<Button buttonType="bulk-delete" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('bulk-delete')).toBe(true)
  })

  it('render Tertiary Outline Button', () => {
    const element = mount(<Button buttonType="tertiary-outline" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('tertiary-outline')).toBe(true)
  })

  it('render Outline Delete Button', () => {
    const element = mount(<Button buttonType="outline-delete" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('outline-delete')).toBe(true)
  })

  it('render Outline Success Button', () => {
    const element = mount(<Button buttonType="outline-success" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('outline-success')).toBe(true)
  })

  it('render Outline Button', () => {
    const element = mount(<Button buttonType="outline" />);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('outline')).toBe(true)
  })
})

describe('With icons', () => {

  it('It should render a button with icon on left side', () => {
    const tree = renderer.create(<Button buttonType="secondary" iconAlignment={"left"} icon={"Save"}>With Icon</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('It should render a button with icon on right side', () => {
    const tree = renderer.create(<Button buttonType="secondary" iconAlignment={"right"} icon={"Save"}>With Icon</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })
});

describe('Disabled button', () => {
  it('It should render a button which is disabled', () => {
    const element = mount(<Button buttonType="primary" disabled={true}>Disabled</Button>);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('Button--state-disabled')).toBe(true);
  })
})

describe('Only Icon', () => {
  it('It should render a button with only icon', () => {
    const tree = renderer.create(<Button buttonType="primary" icon={"Send"} onlyIcon={true} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('With Loader', () => {
  it('It should render a button which is in loading state', () => {
    const element = mount(<Button buttonType="primary" isLoading={true}>Loading</Button>);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('Button--state-loading')).toBe(true);
  })
  it('It should render a button which is in loading state with different loading color', () => {
    const tree = renderer.create(<Button buttonType="secondary" loadingColor={"#6C5CE7"} isLoading={true}>Loading</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('With Button Style Properties', () => {
  it('It should render a button with full width', () => {
    const element = mount(<Button buttonType="primary" isFullWidth={true}>Full Width</Button>);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('Button--style-fullwidth')).toBe(true);
  })
  it('It should render a button as rounded', () => {
    const element = mount(<Button buttonType="primary" isRounded={true}>Full Width</Button>);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('Button--style-circle')).toBe(true);
  })
  it('It should render a button with small size', () => {
    const element = mount(<Button buttonType="primary" size={'small'}>Full Width</Button>);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('Button--size-small')).toBe(true);
  })
  it('It should render a button with large size', () => {
    const element = mount(<Button buttonType="primary" size={'large'}>Full Width</Button>);
    expect(element.find('[data-test-id="cs-button"]').props().className.includes('Button--size-large')).toBe(true);
  })
  it('It should render a button with custom className', () => {
    const tree = renderer.create(<Button buttonType="secondary" className="className-test">Custom ClassName</Button>).toJSON()
    expect(tree.props.className.includes("className-test")).toBe(true);
  })
  it('It should render a button with custom style', () => {
    const CustomStyle = () => {
      return (
        <Button buttonType="secondary" style={{ color: 'red' }}>
          <span>Custom Style</span>
        </Button>
      )
    };
    const tree = renderer.create(<CustomStyle />).toJSON()
    expect(tree).toMatchSnapshot();
  })
})

describe('Button as link', () => {
  it('It should render button as link', () => {
    const element = mount(<Button buttonType="primary" href="https://www.contentstack.com/">Click me</Button>);
    expect(element.find('a').length).toBe(1);
    expect(element.find('button').length).toBe(0);
  });
})

describe('Events and method testing of button', () => {
  it('It should trigger the onClick of the button', () => {
    const mockFunction = jest.fn();
    const ButtononClick = ({ onClickAction }) => {
      return (
        <Button buttonType="primary" onClick={onClickAction}>Publish</Button>
      )
    };
    const element = mount(<ButtononClick onClickAction={mockFunction} />);
    expect(mockFunction).toHaveBeenCalledTimes(0);
    element.find('[data-test-id="cs-button"]').simulate('click');
    expect(mockFunction).toHaveBeenCalledTimes(1);
  });
  it('It should not trigger the onClick of the button when in disabled state', () => {
    const mockFunction = jest.fn();
    const ButtononClick = ({ onClickAction }) => {
      return (
        <Button buttonType="primary" disabled={true} onClick={onClickAction}>Publish</Button>
      )
    };
    const element = mount(<ButtononClick onClickAction={mockFunction} />);
    expect(mockFunction).toHaveBeenCalledTimes(0);
    element.find('[data-test-id="cs-button"]').simulate('click');
    expect(mockFunction).toHaveBeenCalledTimes(0);
  })
})