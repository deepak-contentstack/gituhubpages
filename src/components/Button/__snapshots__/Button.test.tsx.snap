// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Types render Primary Button 1`] = `
<button
  aria-label="aria-button"
  className="Button Button--primary"
  data-test-id="cs-button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div
    className="flex-center"
  >
    <div
      className="flex-v-center"
      style={
        Object {
          "visibility": "visible",
        }
      }
    />
  </div>
</button>
`;

exports[`Only Icon It should render a button with only icon 1`] = `
<button
  aria-label="aria-button"
  className="Button Button--primary Button--icon"
  data-test-id="cs-button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div
    className="flex-center"
  >
    <div
      className="flex-v-center"
      style={
        Object {
          "visibility": "visible",
        }
      }
    />
  </div>
</button>
`;

exports[`With Button Style Properties It should render a button with custom style 1`] = `
<button
  aria-label="aria-button"
  className="Button Button--secondary"
  data-test-id="cs-button"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "color": "red",
    }
  }
  type="button"
>
  <div
    className="flex-center"
  >
    <div
      className="flex-v-center"
      style={
        Object {
          "visibility": "visible",
        }
      }
    >
      <span>
        Custom Style
      </span>
    </div>
  </div>
</button>
`;

exports[`With Loader It should render a button which is in loading state with different loading color 1`] = `
<button
  aria-label="aria-button"
  className="Button Button--secondary Button--state-loading"
  data-test-id="cs-button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div
    className="flex-center"
  >
    <div
      className="Button__loader"
    />
    <div
      className="flex-v-center"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    >
      Loading
    </div>
  </div>
</button>
`;

exports[`With icons It should render a button with icon on left side 1`] = `
<button
  aria-label="aria-button"
  className="Button Button--secondary Button--icon-alignment-left"
  data-test-id="cs-button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div
    className="flex-center"
  >
    <div
      className="flex-v-center"
      style={
        Object {
          "visibility": "visible",
        }
      }
    >
      With Icon
    </div>
  </div>
</button>
`;

exports[`With icons It should render a button with icon on right side 1`] = `
<button
  aria-label="aria-button"
  className="Button Button--secondary Button--icon-alignment-right"
  data-test-id="cs-button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div
    className="flex-center"
  >
    <div
      className="flex-right"
      style={
        Object {
          "visibility": "visible",
        }
      }
    >
      With Icon
    </div>
  </div>
</button>
`;
